src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -13,8 +13,12 @@ class PromotionRepository extends EntityRepository
     /**
      * @return array[]
      */
-    public function findActivePromotionsByContext(int $contextId, DateTimeImmutable $datetime, ?int $limit, int $offset): array
+    public function findActivePromotionsByContext(array $ancestryIds, DateTimeImmutable $datetime, ?int $limit, int $offset): array
     {
+        // $contextId is the current context. We want to get both cascading and non-cascading promos for this id
+        // $ancestryIds is an array ids of the parents of the current context. For these ids we only want cascading promos
+        $contextId = array_pop($ancestryIds);
```

Comment:
What if someone is manually building the array to send through and mucks it up so that it ends up empty?